frontend/model/contracts/group.js

```diff
@@ -1360,13 +1360,81 @@ sbp('chelonia/defineContract', {
         const current = state.settings?.distributionDate
-        if (current !== period) {
-          // right before updating to the new distribution period, make sure to update various payment-related group streaks.
-          updateGroupStreaks({ state, getters })
-          state.settings.distributionDate = period
+        const currentPeriodPayments = cloneDeep(getters.groupPeriodPayments[current])
+
+        if (comparePeriodStamps(periodTo, current) > 0) {
+          const getAllEmptyPeriodsBetween = () => {
+            const periods = []
+            let recorded = current
+            while (true) {
+              const period = plusOnePeriodLength(recorded, periodLength)
+              if (comparePeriodStamps(periodTo, period) > 0) {
+                periods.push(period)
+                recorded = period
+              } else {
+                break
+              }
+            }
+            return periods
+          }
+
+          // There can be one or more distribution periods without any payments (empty-periods) between current 'distributionDate'
+          // and the distribution period to update to. (eg. No one in the group hasn't logged in for a long time)
+          // 'paymentsByPeriod' object in the group state and the payment-related group streaks need to be populated/updated accordingly for those empty periods in this case.
+          // (Reference issue: https://github.com/okTurtles/group-income/issues/2982)
+          const emptyPeriodsBetweenCurrentAndTo = getAllEmptyPeriodsBetween()
+          const allPeriodsToUpdate = [
+            ...emptyPeriodsBetweenCurrentAndTo,
+            periodTo
+          ]
+
+          allPeriodsToUpdate.forEach((period, index) => {
+            initFetchPeriodPayments({ contractID, meta, periodTo: period, state, getters })
+          })
+
+          // There can be two cases:
+          // 1) The period to update comes right after current period
+          // 2) There are one or more empty periods (periods with no payments made) between current period and the period to update.
+          // In case of 1), we add +1 to the payment-related group streaks.
+          // In case of 2), those streaks need to be reset except for 'missedPayments' streak.
+          if (emptyPeriodsBetweenCurrentAndTo.length === 0) {
+            updateGroupStreaks({ state, getters })
+          } else {
+            state.streaks = {
+              ...state.streaks,
+              ...({
+                lastStreakPeriod: null,
+                fullMonthlyPledges: 0,
+                fullMonthlySupport: 0,
+                onTimePayments: {}
+              })
+            }
+
+            // Add the number of empty periods to 'missedPayments' streaks
+            const groupHasReceivers = Object.keys(getters.groupReceiverProfiles).length > 0
+
+            if (groupHasReceivers) {
+              const allPledgerIds = Object.keys(getters.groupPledgerProfiles)
```

Comment:
Wouldn't it make sense to check if we have both receivers and pledgers? As we haven't missed a payment if no one was planning to make one

Reply:
It's safe to leave it as is because when there is no pledgers, `allPledgerIds` in below block becomes an empty array and it skips the logic execution:

```
for (const pledgerId of allPledgerIds) {
...
}
```

But will add a check for pledgers here too to make things clear.